src/Voron/Data/Tables/TableSchema.cs

```diff
@@ -61,26 +53,25 @@ static TableSchema()
             }
         }
 
-        public SchemaIndexDef Key => _primaryKey;
+        public StaticBTreeIndexDef Key => _primaryKey;
 
         // Indexes are conceptually Dictionary<index name, Dictionary<unique index value, HashSet<storage id>>
 
         /// <summary>
         /// Indexes are conceptually Dictionary&lt;index name, Dictionary&lt;unique index value, HashSet&lt;storage id&gt;&gt;
         /// </summary>
-        public Dictionary<Slice, SchemaIndexDef> Indexes => _indexes;
+        public Dictionary<Slice, AbstractBTreeIndexDef> Indexes => _indexes;
 
         // FixedSizeIndexes are conceptually Dictionary<index name, Dictionary<long value, storage id>>
 
         /// <summary>
         /// FixedSizeIndexes are conceptually Dictionary&lt;index name, Dictionary&lt;long value, storage id&gt;&gt;
         /// </summary>
-        public Dictionary<Slice, FixedSizeSchemaIndexDef> FixedSizeIndexes => _fixedSizeIndexes;
+        public Dictionary<Slice, FixedSizeTreeIndexDef> FixedSizeIndexes => _fixedSizeIndexes;
 
-        public class SchemaIndexDef
+        public class StaticBTreeIndexDef : AbstractBTreeIndexDef
```

Comment:
I like that the indexes classes are nested inside `TableSchema` so we have usages like `new TableSchema.IndexDef()`. Although we could put them in separate files and use partials. So in `TableSchema.IndexDef.cs` we can have:

```
public partial TableSchema
{
  public class IndexDef
  {
  }
}
```